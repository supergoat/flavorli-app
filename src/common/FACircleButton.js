// @flow
import React, { Component } from "react";
import {
  Text,
  TouchableOpacity
} from "react-native";
import type {
   ViewStyleProp
} from 'react-native/Libraries/StyleSheet/StyleSheet';
import FAColors from "../common/FAColors";
import FAStyleSheet from "../common/FAStyleSheet";

/** ============================================================================
<FACircleButton
  text={string}
  disabled={boolean}
/>
--------------------------------------------------------------------------------
A flavorli themed circular button
============================================================================= */
type Props = {
  text: string,
  disabled?: boolean
};
class FACircleButton extends Component<Props> {
  render() {
    const { text, disabled, ...rest } = this.props;

    return (
      <TouchableOpacity
        {...rest}
        disabled={disabled}
        activeOpacity={0.8}
        style={[
          styles.button,
          disabled && styles.disabled
        ]}
      >
        <Text style={styles.buttonText}>{text}</Text>
      </TouchableOpacity>
    );
  }
}

/* StyleSheet
============================================================================= */
const styles = FAStyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 38,
    height: 38,
    borderRadius: 19,
    borderWidth: 1,
    borderColor: FAColors.oxfordBlue
  },
  buttonText: {
    fontSize: 25,
    color: FAColors.oxfordBlue,
    fontFamily: 'Lato',
    fontWeight: 'bold'
  },
  disabled: {
    opacity: 0.5
  }
});

/* Export
============================================================================= */
export default FACircleButton;
