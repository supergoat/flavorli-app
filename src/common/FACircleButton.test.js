// @flow
import React from 'react';
import { View, Text } from 'react-native';
import FACircleButton from './FACircleButton';
import renderer from 'react-test-renderer';

describe('FACircleButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <FACircleButton
        text='button'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when disabled is true', () => {
    const tree = renderer.create(
      <FACircleButton
        text='button'
        disabled
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
