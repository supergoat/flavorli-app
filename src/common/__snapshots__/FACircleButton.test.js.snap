// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FACircleButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#333C4E",
      "borderRadius": 19,
      "borderWidth": 1,
      "height": 38,
      "justifyContent": "center",
      "opacity": 1,
      "width": 38,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333C4E",
        "fontFamily": "Lato",
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  >
    button
  </Text>
</View>
`;

exports[`FACircleButton renders correctly when disabled is true 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#333C4E",
      "borderRadius": 19,
      "borderWidth": 1,
      "height": 38,
      "justifyContent": "center",
      "opacity": 0.5,
      "width": 38,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333C4E",
        "fontFamily": "Lato",
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  >
    button
  </Text>
</View>
`;
